### PR TITLE
Transpose polysets

### LINKED ISSUE #395: Transpose polysets
Many operations in polyset.cpp could be done on contiguous memory if the tables `P` were indexed `P(polynomial_index, derivative_index, point_index)`.

Currently the indexing is P(derivative_index, point_index, polynomial_index)`.

### PR DESCRIPTION
**Pull request by mscroggs**
Transpose the last two indices of polysets, so that `xt::row` is now used instead of `xt::col`

Fixes #395.

### CODE REVIEW

File: cpp/basix/finite-element.cpp [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #include <basix/version.h>
 #include <numeric>
 #include <xtensor/xbuilder.hpp>
+#include <xtensor/xio.hpp>
```
Comment (jorgensd) [inline cpp/basix/finite-element.cpp:22]: This should be removed

File: cpp/basix/lattice.cpp [hunk 2]
```diff
@@ -156,14 +156,13 @@ xt::xtensor<double, 2> tabulate_dlagrange(int n,
   xt::xtensor<double, 3> dual_values
       = polyset::tabulate(cell::type::interval, n, 0, equi_pts);
   xt::xtensor<double, 2> dualmat({dual_values.shape(2), dual_values.shape(1)});
-  dualmat.assign(xt::transpose(xt::view(dual_values, 0, xt::all(), xt::all())));
+  dualmat.assign(xt::view(dual_values, 0, xt::all(), xt::all()));
 
   xt::xtensor<double, 3> tabulated_values
       = polyset::tabulate(cell::type::interval, n, 0, x);
   xt::xtensor<double, 2> tabulated(
       {tabulated_values.shape(2), tabulated_values.shape(1)});
```
Comment (jorgensd) [inline cpp/basix/lattice.cpp:164]: Is this correct when everything else has been transposed?

Comment (chrisrichardson) [reply cpp/basix/lattice.cpp:164]: I think it's ok

File: cpp/basix/polyset.cpp [hunk 3]
```diff
@@ -80,15 +80,15 @@ void tabulate_polyset_line_derivs(xt::xtensor<double, 3>& P, std::size_t n,
...
+      xt::row(result, p) = (x0 * 2.0 - 1.0) * xt::row(result, p - 1) * (a + 1.0)
+                           + 2 * k * xt::row(result0, p - 1) * (a + 1.0)
+                           - xt::row(result, p - 2) * a;
     }
   }
 
   // Normalise
   for (std::size_t p = 0; p <= n; ++p)
-    xt::view(P, xt::all(), xt::all(), p) *= std::sqrt(2 * p + 1);
+    xt::view(P, xt::all(), p, xt::all()) *= std::sqrt(2 * p + 1);
```
Comment (IgorBaratta) [inline cpp/basix/polyset.cpp:91]: I'm not big fan of xtensor's broadcasting.
Shall we use plain loops instead? Maybe in a different PR?

**Review by IgorBaratta — APPROVED** [review]

**Pull request merged**